src/chai/ManagedArray_thin.inl

```diff
@@ -194,16 +194,29 @@ CHAI_HOST void ManagedArray<T>::reallocate(size_t new_elems)
     T* new_ptr;
 
   #if defined(CHAI_ENABLE_UM)
-    gpuMallocManaged(&new_ptr, sizeof(T) * new_elems);
-    gpuMemcpy(new_ptr, m_active_pointer, sizeof(T) * m_elems, gpuMemcpyDefault);
+    if (new_elems > 0) {
+       gpuMallocManaged(&new_ptr, sizeof(T) * new_elems);
+       gpuMemcpy(new_ptr, m_active_pointer, sizeof(T) * m_elems, gpuMemcpyDefault);
+    }
     gpuFree(m_active_pointer);
   #else  // not CHAI_ENABLE_UM
-    new_ptr = static_cast<T*>(realloc(m_active_pointer, sizeof(T) * new_elems));
+    if (new_elems > 0) {
+       new_ptr = static_cast<T*>(realloc(m_active_pointer, sizeof(T) * new_elems));
+    }
+    else {
+       ::free((void *)m_active_pointer);
```

Comment:
Could set new_ptr to nullptr here, and avoid the changes starting at line 212.

Reply:
Should I set it here plus one for the CHAI_ENABLE_UM branch?  Or give new_ptr a null initializer?  I'm open to anything.... don't know what the best style is here.

Reply:
The changes for CHAI_ENABLE_UM are not strictly required, but it seemed like the analogous change.

Reply:
I think I would give new_ptr a nullptr initializer.